experiments/kube-aws/az-down/experiment/az-down.go

```diff
@@ -0,0 +1,329 @@
+				msg := "AUT: Running, Probes: Unsuccessful"
+				types.SetEngineEventAttributes(&eventsDetails, types.PreChaosCheck, msg, "Warning", &chaosDetails)
+				events.GenerateEvents(&eventsDetails, clients, &chaosDetails, "ChaosEngine")
+				result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+				return
+			}
+			msg = "AUT: Running, Probes: Successful"
+		}
+		// generating the events for the pre-chaos check
+		types.SetEngineEventAttributes(&eventsDetails, types.PreChaosCheck, msg, "Normal", &chaosDetails)
+		events.GenerateEvents(&eventsDetails, clients, &chaosDetails, "ChaosEngine")
+	}
+
+	// start of experiment logic
+	// TODO create replicated topic in kafka
+
+	// Configure AWS Credentials
+	if err = aws.ConfigureAWS(); err != nil {
+		log.Errorf("AWS authentication failed, err: %v", err)
+		failStep := "Configure AWS configuration (pre-chaos)"
+		result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+		return
+	}
+
+	// create ec2 client
+	ec2Svc := aws.GetNewEC2Client(&experimentsDetails)
+
+	// Get all instances in region that belong to our cluster under test
+	clusterInstances, err := aws.GetClusterInstancesInRegion(&ec2Svc, experimentsDetails.ClusterIdentifier)
+	if err != nil {
+		log.Errorf("failed to get the ec2 instance instances in region, err: %v", err)
+		failStep := "Getting all instances in region (pre-chaos)"
+		result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+		return
+	}
+
+	// select az to be targeted
+	azToTarget := getAzToTarget(clusterInstances)
+
+	// get subnets for target az
+	azName := fmt.Sprintf("ZoneName: %s", azToTarget)
+	azNameFilter := ec2.Filter{Values: []*string{&azName}}
+	filters := []*ec2.Filter{&azNameFilter}
+
+	subnetsDescription, err := ec2Svc.DescribeSubnets(&ec2.DescribeSubnetsInput{Filters: filters})
+
+	// gather subnet details required
+	azDetails := experimentTypes.AzDetails{
+		AzName:  azName,
+	}
+	subnetDetails := []*experimentTypes.Subnet{}
+	for _, subnet := range subnetsDescription.Subnets {
+		subnetDetail := experimentTypes.Subnet{
+			Id: *subnet.SubnetId,
+			VpcId: *subnet.VpcId,
+		}
+		subnetDetails = append(subnetDetails, &subnetDetail)
+	}
+	azDetails.Subnets = subnetDetails
+
+	// create and assign dummy acl to each subnet vpc
+	for i, subnet := range azDetails.Subnets {
+		output, err := ec2Svc.CreateNetworkAcl(&ec2.CreateNetworkAclInput{VpcId: &subnet.VpcId})
+		if err != nil {
+			log.Errorf("Failed to assign dummy ACM to vpc of az being targeted, err: %v", err)
+			failStep := "Failed to assign dummy ACL to vpc of az being targeted"
+			result.RecordAfterFailure(&chaosDetails, &resultDetails, failStep, clients, &eventsDetails)
+			return
+		}
+		subnet.DummyAclId = fmt.Sprint(output.NetworkAcl.NetworkAclId)
+		azDetails.Subnets[i] = subnet
+		log.Info(fmt.Sprintf("New ACL created and assigned to vpc id: %s.\n%s", subnet.VpcId, output))
```

Comment:
I think there is no need for Spintf here, logging can take multiple parameters to print/log.